sirepo/uri.py

```diff
@@ -28,17 +22,21 @@
 PATH_INFO_CHAR = '*'
 
 
-def app_root(sim_type):
+def app_root(sim_type, external=False):
     """Generate uri for application root
 
     Args:
         sim_type (str): application name
+        external (bool): external uri? [False]
 
     Returns:
         str: formatted URI
     """
-    t = http_request.sim_type(sim_type)
-    return ROOT if t is None else ROOT + t
+    return uri_router.uri_for_api(
+        'root',
+        params=PKDict(path_info=http_request.sim_type(sim_type) or ''),
```

Comment:
Now that we are not constructing the uri here, this call should be:

```py
t = http_request.sim_type(sim_type)
return uri_router.uri_for_api(
        'root',
        params=PKDict(path_info=t) if t else None,
        external=external,
)
```

Reply:
ad61fc813610afe4d4c57cbe0e4322c503d6ac6c